docs/sphinx/examples/cpp/algorithms/grover.cpp

```diff
@@ -40,7 +40,14 @@ struct oracle {
   }
 };
 
+// Entry-point kernel running the Grover's algorithm with a specific oracle.
+struct run_grover {
+  __qpu__ auto operator()(const int n_qubits, const int n_iterations) {
+    grover_algorithm{}(n_qubits, n_iterations, oracle{});
+  }
+};
+
 int main() {
-  auto counts = cudaq::sample(run_grover{}, 3, 1, oracle{});
+  auto counts = cudaq::sample(run_grover{}, 3, 1);
```

Comment:
From commit: https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/d8a448837a298b728141fdad6a62ac9d54ac58df
> Don't pass kernels as arguments to entry-point invocation.

Does this mean that kernels can no longer be passed as arguments to entry-point functions when called with `cudaq::sample()`? If so, is that consistent with the language specification? @amccaskey

Reply:
> Does this mean that kernels can no longer be passed as arguments to entry-point functions when called with cudaq::sample()?

Only when the user wants to use the remote REST platform target. It will work as usual for all other targets. Since these examples are validated for all targets, @bettinaheim suggested that we could rewrite them to be more conservative in terms of entry-point arguments. Also, I think callable is not listed in the entry-point kernel argument types yet.

Reply:
These changes to examples were reverted in https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/44f247bb3878916f2bc7c22297023963ea5ad16e. 

Adding a separate test for error message for this case (callable as an argument to entry point kernel for remote simulator in library mode) in https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/e38d1418421d5df66cb47e4c0028ac7fb4755617.